### PR TITLE
Mouse changes

### PR DESCRIPTION
**Pull request by ghost**
Turning them from minor scurries to 'Janitor get the fucking mouse traps!'

- [x] Mice actually consume food
- [x] Add downsides to spam-eating. Fat rats were suggested
- [x] Mice are attracted to food, whether they like it or not. Meaning you can bait mousetraps
- [x] Mice can nibble at exposed wires, likely to electrocute them, and that's it.
- [x] Got N3X15 to get angry over one of my PRs
- [x] Mice can carry diseases
- [x] Rewrite most of the infection code and go through my stock of beer so mice can carry diseases

:cl: 
* rscadd: Mice can now eat food properly, making them a proper pest to chefs.
* rscadd: Mice can now grow fat off eating food, making them farmable by chefs and unable to ventcrawl.
* rscadd: Mice can now starve if they don't eat food.
* rscadd: Mice can now 'nibble' at wires
* rscadd: two subtypes of mouse, one can carry and transmit diseases, the other can cut wires through nibbling (and potentially zap itself when doing so)

### CODE REVIEW

**Comment by gbasood** [comment]
Hmm
Mice are possessable at any time by a ghost, giving them the ability to chew through wire might not be great
On the other hand, as long as it's logged, it's just a typical admin problem

**Comment by Intigracy** [comment]
Well, it has a chance to shock them instead, plus mice have cooldowns on becoming a mouse if you die as one.

**Comment by ghost** [comment]
Should be logged the same way as it would with wirecutters

**Comment by AutismSpider** [comment]
five min cool down so it take lot of ghost to start rat pack

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 1]
```diff
@@ -53,6 +54,21 @@
...
+		if(!food_target)
+			for(var/obj/item/weapon/reagent_containers/food/snacks/C in can_see)
+				food_target = C
+				break
+		if(!(food_target in can_see))
+			food_target = null
+		if(food_target)
+			if(Adjacent(food_target))
+				food_target.attack_animal(src)
+			step_towards(src, food_target)
```
Comment (Intigracy) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: Won't food_target be null if attack_animal finishes off the food?

Comment (Intigracy) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm]: Meaning they'll eat the food from next to the mouse trap and not move onto it.

Comment (ghost) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm]: Fair point, I'll have the step_towards before the adjacent check next chance I get

**Comment by N3X15** [comment]
No, having mice impact the round is a fucking terrible idea.

**Comment by ghost** [comment]
`// FUCK mice. - N3X`

**Comment by Shadowmech88** [comment]
You mean player-controlled mice, or mice altogether?

**Comment by ghost** [comment]
Mice altogether, player controlled mice are the only ones that can chew wires however

**Comment by Shadowmech88** [comment]
>player controlled mice are the only ones that can chew wires however

That's worse than it just being a general mouse ability.
And the ability to have a chance to cut power to areas of the station is a tad bit strong for ghosts to have, I think.

**Comment by TheFriggenSawbones** [comment]
agreed on the chewing wires thing, it sounds scary

**Comment by ghost** [comment]
Mice being able to cut wires by chewing them is now removed.

**Comment by Lokiamis** [comment]
Say goodbye to your emitter wires :^)

**Comment by d3athrow** [comment]
nonantag mice when?

**Comment by ghost** [comment]
Right, I'll leave this for the emojiocracy to decide

Thumbs up for letting mice transmit diseases, thumbs down for please no, confused for meme option

**Comment by Lokiamis** [comment]
Player controlled pet mouse/rat that can spread disease, chew through wires, and bite people as a traitor item for virologists.

**Comment by TheFriggenSawbones** [comment]
Being able to transmit viruses as a player controlled mouse sounds bad.

**Comment by JustSumBody** [comment]
Agreed with nexis
MAYBE the food eating thing, but the virus and wirecutting is pretty broken for something that can be re-spawned from a salty ghost. 
Also consider yeah, this is basically giving mice antag status. Which I'm not keen on for the previous salty ghost reason.

**Comment by jakmak** [comment]
Everything here is fine in my opinion EXCEPT the wirecutting. For the love of all that is holy DO NOT let mice (npc or player controlled ) be able to cut cables.

**Comment by Exxion** [comment]
He removed that many comments ago. Keep up.

**Comment by aceedex** [comment]
nothing wrong with wire cutting. just pay attention to the APCs

**Comment by SonixApache** [comment]
THE PLAGUE

**Comment by ghost** [comment]
Okay, it has been 13 hours now, I'll leave it up to 24 for the conclusion of that vote

In the meantime, should I make a virologist traitor item for plague rats?

**Comment by murnandreams** [comment]
Maybe an injector to put into mice, that gives em virus transmission, wire chewing, and pretty much harmless bites. Giving them some sort of loyalty to the virologist may be important. Hell, if you add in a deadchat alert, it can even work on non-player mice. Just some ideas.

**Comment by ghost** [comment]
I was thinking of having something of a cross between monkey cubes and borer eggs.

Imprint ID of the user on the cube, wet the rat cube, it'll ask deadchat for recruitment. If none are available it'll fail and wait some time.

**Comment by murnandreams** [comment]
An afterthought: maybe even have it be somewhat transmitable between mice. Make janitor with mousetraps a necessity.

**Comment by ghost** [comment]
That was the intention of the PR to begin with, and for engineers to keep their wires under control.

But, defficiency kinda puts a hole in that, as usual

**Comment by Kurfursten** [comment]
Actually having mice chew wires all the time might train engineers to actually look for broken wires for once. But the real trouble is that if a mouse camps near the airlocks to the singulo and waits for someone to exit, they could quickly sneak in after the person leaves, eat the wire, and loose.

Having AI mice or "rats" eat wires isn't actually a terrible idea by any means, but players are just too crafty for a respawning role to have that much power.

**Comment by murnandreams** [comment]
Well, the singuloosing can be fixed really easily. Just add plating to the map, in the airlock.

**Comment by ghost** [comment]
Alright, so how about a client check to see if a mouse has no client, and if so, every so often it may try to chew a wire in its vicinity?

There's plenty of NPC mice in maint that could cause trouble with this, however, and our power grids are kinda linear rather than having backup routes

**Comment by N3X15** [comment]
@Kurfursten 
>Actually having mice chew wires all the time might train engineers to actually look for broken wires for once. 

After engineers are done with power, they vanish into the aether or are wrecked by antags for their suits.  Expecting them to do anything about wires is hilariously naive when half of them couldn't wire to save their lives and others wouldn't know what to do with a T-ray scanner if it had an instruction manual with pictures.

**Comment by Kurfursten** [comment]
@N3X15 That's how they are right now because there's no upkeep on the system. They've essentially been trained to vanish into the ether.

I used to play engineer a ton, and your assessment is of course correct, I could pretty much count on two hands every single engineer who knew how to use a multitool on a wire to see if it was powered. But it became that way because there was never any need for them to learn. Most engineers can at least set up one engine because it's the necessity of their job, but by never requiring engineers to learn to find wire breaks, we train them to not gain that expertise.

Like I said, it wouldn't really work in the context of player mice since they can respawn every five minutes and it would make every round hell, but it might be an interesting element to bring into the 'mice breeding' event which is quite rare. (The game has to choose the breeding event AND choose a place with plating + wires AND choose to spawn mice instead of slimes or spiders or borers or bats)

**Comment by Probe1** [comment]
Engineering for experienced Engineers has gotten very little content over the last 3 years.

Engineering has gotten a lot of great quality of life content.

But as far as things to do not so much.

Also all of the fun time damage that the proliferation of Max Cap Bombs has done

File: code/modules/virus2/helpers.dm [hunk 2]
```diff
@@ -1,35 +1,38 @@
...
-			if(istype(M:wear_suit, /obj/item/clothing/suit/space))
+			if(istype(H:wear_suit, /obj/item/clothing/suit/space))
 				score += 35
-			if(istype(M:wear_suit, /obj/item/clothing/suit/bio_suit))
+			if(istype(H:wear_suit, /obj/item/clothing/suit/bio_suit))
 				score += 35
 
-//	log_debug("[M]'s resistance to [vector] viruses: [score]")
-
+	log_debug("[M]'s resistance to [vector] viruses: [score]")
```
Comment (ghost) [inline code/modules/virus2/helpers.dm]: Whoops, forgot to recomment these out

File: code/modules/virus2/helpers.dm [hunk 3]
```diff
@@ -1,35 +1,38 @@
...
+	if(istype(M, /mob/living/simple_animal/mouse))
+		var/mob/living/simple_animal/mouse/MM = M
+		if(MM.disease_carrier)
+			return 1
 	if(prob((min(score, 100) - 100) ** 2 / 100))
-	//	log_debug("Infection got through")
+		log_debug("Infection got through")
```
Comment (ghost) [inline code/modules/virus2/helpers.dm]: This one too

File: code/modules/virus2/helpers.dm [hunk 4]
```diff
@@ -49,17 +52,19 @@ proc/airborne_can_reach(turf/source, turf/target, var/radius=5)
 //Attemptes to infect mob M with virus. Set forced to 1 to ignore protective clothing.  Returns 1 if successful.
 /proc/infect_virus2(var/mob/living/carbon/M,var/datum/disease2/disease/disease,var/forced = 0, var/notes="")
 	if(!istype(disease))
-//		log_debug("Bad virus")
+		log_debug("Bad virus")
```
Comment (ghost) [inline code/modules/virus2/helpers.dm]: And this one

File: code/modules/virus2/helpers.dm [hunk 5]
```diff
@@ -49,17 +52,19 @@ proc/airborne_can_reach(turf/source, turf/target, var/radius=5)
...
 		return 0
-	if(!istype(M))
-//		log_debug("Bad mob")
+	if(!istype(M, /mob/living/carbon) && !istype(M, /mob/living/simple_animal/mouse))
```
Comment (ghost) [inline code/modules/virus2/helpers.dm]: Use the can_be_infected helper

File: code/modules/virus2/helpers.dm [hunk 6]
```diff
@@ -49,17 +52,19 @@ proc/airborne_can_reach(turf/source, turf/target, var/radius=5)
...
+		log_debug("Bad mob")
```
Comment (ghost) [inline code/modules/virus2/helpers.dm]: And this one

**Comment by ghost** [comment]
Right

The good news

- Mice can now carry diseases, and infect others around them
- They will passively walk towards and invisibly nibble humans, don't worry, this goes through the proper biochecks so no monkey madness
- Disease carrying and damaging cables are now controlled via booleans that are default to off, there are subtypes that can be spawned that have these on

The bad news

- The only way you can start a rat being infected in a 'controlled' environment is to smash a virus plate off their head, or have somebody who's ill go near them
- Not entirely sure if I broke anything
- If you spawn a bunch of wire nibblers, one will work towards sacrificing itself so the survivors can go to town

plague_rats

it begiiins

manical_laughter

cable_biting_flaw

File: code/modules/virus2/effect/effect.dm [hunk 7]
```diff
@@ -230,8 +230,8 @@
 
 /datum/disease2/effect/cough/activate(var/mob/living/carbon/mob)
 	mob.say("*cough")
-	for(var/mob/living/carbon/M in oview(2,mob))
-		mob.spread_disease_to(M)
+	for(var/mob/living/M in oview(2,mob))
+		spread_disease_to(src, M)
```
Comment (ghost) [inline code/modules/virus2/effect/effect.dm]: Check through can_be_infected

**Comment by ghost** [comment]
I think this is about done from what I can see, just needs review

**Comment by N3X15** [comment]
Merging with 10 yes, 4 no, as it's currently Monday, 12AM GMT.

**Comment by ghost** [comment]
Well, it hasn't been reverted yet, which is promising

**Pull request merged**